p2p/agent.go

```diff
@@ -502,8 +502,9 @@ func (p *agent) connect(ctx context.Context) error {
 	connErrChan := make(chan error, len(p.cfg.BootstrapNodes))
 
 	// try to connect to all bootstrap node beside itself.
-	for _, bootAddr := range p.bootNodeAddr {
+	for i := range p.bootNodeAddr {
```

Comment:
```
for _, bootAddr := range p.bootNodeAddr {
```
`bootAddr` is a single variable defined before the loop, so its value is overwritten by the last element in the array. This is usually OK if the loop executes sequentially -- but here a goroutine is started for each loop, so all goroutines end up using the last element in the array